src/server/game/Globals/ObjectMgr.cpp

```diff
@@ -3557,65 +3557,65 @@ void ObjectMgr::LoadPlayerInfo()
-                    continue;
-                }
-
-                ChrRacesEntry const* rEntry = sChrRacesStore.LookupEntry(current_race);
-                if (!rEntry)
-                {
-                    LOG_ERROR("sql.sql", "Wrong race {} in `playercreateinfo` table, ignoring.", current_race);
-                    continue;
-                }
-
-                if (current_class >= MAX_CLASSES)
+                if (raceMask != 0 && !(raceMask & RACEMASK_ALL_PLAYABLE))
                 {
-                    LOG_ERROR("sql.sql", "Wrong class {} in `playercreateinfo` table, ignoring.", current_class);
+                    LOG_ERROR("sql.sql", "Wrong race mask {} in `playercreateinfo` table, ignoring.", raceMask);
                     continue;
                 }
 
-                if (!sChrClassesStore.LookupEntry(current_class))
+                if (classMask != 0 && !(classMask & CLASSMASK_ALL_PLAYABLE))
                 {
-                    LOG_ERROR("sql.sql", "Wrong class {} in `playercreateinfo` table, ignoring.", current_class);
+                    LOG_ERROR("sql.sql", "Wrong class mask {} in `playercreateinfo` table, ignoring.", classMask);
                     continue;
                 }
 
                 // accept DB data only for valid position (and non instanceable)
                 if (!MapMgr::IsValidMapCoord(mapId, positionX, positionY, positionZ, orientation))
                 {
-                    LOG_ERROR("sql.sql", "Wrong home position for class {} race {} pair in `playercreateinfo` table, ignoring.", current_class, current_race);
+                    LOG_ERROR("sql.sql", "Wrong home position for class mask {} race mask {} pair in `playercreateinfo` table, ignoring.", classMask, raceMask);
                     continue;
                 }
 
                 if (sMapStore.LookupEntry(mapId)->Instanceable())
                 {
-                    LOG_ERROR("sql.sql", "Home position in instanceable map for class {} race {} pair in `playercreateinfo` table, ignoring.", current_class, current_race);
+                    LOG_ERROR("sql.sql", "Home position in instanceable map for class mask {} race mask {} pair in `playercreateinfo` table, ignoring.", classMask, raceMask);
                     continue;
                 }
 
                 PlayerInfo* info = new PlayerInfo();
-                info->mapId = mapId;
-                info->areaId = areaId;
-                info->positionX = positionX;
-                info->positionY = positionY;
-                info->positionZ = positionZ;
-                info->orientation = orientation;
-                info->displayId_m = rEntry->model_m;
-                info->displayId_f = rEntry->model_f;
-                _playerInfo[current_race][current_class] = info;
-
-                ++count;
+                for (uint32 raceIndex = RACE_HUMAN; raceIndex < MAX_RACES; ++raceIndex)
+                {
+                    if (raceMask == 0 || ((1 << (raceIndex - 1)) & raceMask))
```

Comment:
if (!raceMask ||

Reply:
On top of changing `(raceMask == 0) -> (!raceMask)`, I also changed instances of `(raceMask != 0) -> (raceMask)` since they seemed (to me) to be similar.

Reply:
yep